### PR TITLE
chore: implement route53 customization for hostzone trimming

### PR DESCRIPTION
**Pull request by wooj2**
dependent on:
https://github.com/awslabs/smithy-swift/pull/397
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codegen/sdk-codegen/aws-models-test/route-53/route-53-tests.smithy [hunk 1]
```diff
@@ -0,0 +1,20 @@
+$version: "1.0"
+
+namespace com.amazonaws.route53
+
+use smithy.test#httpRequestTests
+
+
+apply ListResourceRecordSets @httpRequestTests([
+    {
+        id: "ListResourceRecordSetsTrimHostdZone",
```
Comment (kneekey23) [inline codegen/sdk-codegen/aws-models-test/route-53/route-53-tests.smithy:10]: nit: sp ListResourceRecordSetsTrimHostdZone -> ListResourceRecordSetsTrimHostedZone

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/Route53TrimHostedZone.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import software.amazon.smithy.model.traits.HttpLabelTrait
+import software.amazon.smithy.model.transform.ModelTransformer
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.integration.middlewares.handlers.MiddlewareShapeUtils
+import software.amazon.smithy.swift.codegen.middleware.OperationMiddleware
+import software.amazon.smithy.swift.codegen.model.hasTrait
+
+internal val Route53ShapeId: ShapeId = ShapeId.from("com.amazonaws.route53#AWSDnsV20130401")
```
Comment (kneekey23) [inline codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/Route53TrimHostedZone.kt:17]: could this be private? any reason for it to be internal if it's only used in this one file?

**Review by kneekey23 — APPROVED** [review]
LGTM, one nit and one question non blocking

File: codegen/sdk-codegen/aws-models-test/route-53/route-53-tests.smithy (hunk 1 above)
Comment (wooj2) [reply codegen/sdk-codegen/aws-models-test/route-53/route-53-tests.smithy:10]: i'll update and notify rust to update theirs

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/Route53TrimHostedZone.kt (hunk 2 above)
Comment (wooj2) [reply codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/Route53TrimHostedZone.kt]: updated, thanks!

**Pull request merged**